core/src/main/java/com/microsoft/applicationinsights/channel/concrete/localforwarder/LocalForwarderTelemetryChannel.java

```diff
@@ -34,6 +34,9 @@ public LocalForwarderTelemetryChannel(Map<String, String> namesAndValues) {
     @Override
     protected synchronized void initialize(String configurationFileEndpoint, String maxTransmissionStorageCapacity, boolean developerMode,
                                            LimitsEnforcer maxTelemetryBufferCapacityEnforcer, LimitsEnforcer sendIntervalInSeconds, boolean throttling, int maxInstantRetry) {
+        if (isInitailized) {

```

Comment:
I thought that since this variable is not atomic we used double locking no?

Reply:
No, only check-lock-check needs volatile. A synchronized method is always safe. Plus this is just called once so we don't get any benefit from the double check pattern.